timely_beliefs/visualization/graphs.py

```diff
@@ -0,0 +1,283 @@
+from typing import Tuple, Union
+
+import altair as alt
+
+from timely_beliefs.visualization import selectors
+
+
+def base_chart(source, belief_horizon_unit: str) -> alt.Chart:
+    return (
+        alt.Chart(source)
+        .encode(
+            x=alt.X(
+                "event_start",
+                axis=alt.Axis(labelFlush=False),
+                scale={"domain": selectors.time_selection_brush.ref()},
+                title="Event start",
+            ),
+            color=selectors.source_color_or(selectors.idle_color),
+            tooltip=[
+                alt.Tooltip(
+                    "event_start:T",
+                    timeUnit="yearmonthdatehoursminutes",
+                    title="Event start",
+                ),
+                alt.Tooltip(
+                    "event_end:T",
+                    timeUnit="yearmonthdatehoursminutes",
+                    title="Event end",
+                ),
+                alt.Tooltip("expected_value:Q", title="Expected value", format=".2f"),
+                alt.Tooltip(
+                    "belief_time:T",
+                    timeUnit="yearmonthdatehoursminutes",
+                    title="Belief time",
+                ),
+                alt.Tooltip(
+                    "belief_horizon:Q",
+                    title="Belief horizon (%s)" % belief_horizon_unit,
+                ),
+                alt.Tooltip("source", title="Source"),
+            ],
+        )
+        .properties(width=550, height=200)
+    )
+
+
+def time_series_chart(
+    base,
+    active_fixed_viewpoint_selector: bool,
+    sensor_name: str,
+    sensor_unit: str,
+    belief_horizon_unit: str,
+    intuitive_forecast_horizon: bool,
+    interpolate: bool,
+    ci: float,
+    event_value_range: Tuple[float, float],
+) -> alt.LayerChart:
+
+    # Configure the stepwise line for the reference
+    if interpolate is True:
+        ts_line_reference_chart = base.mark_line(interpolate="monotone")
+    else:
+        ts_line_reference_chart = base.mark_rule().encode(x2=alt.X2("event_end:T"))
+    ts_line_reference_chart = ts_line_reference_chart.encode(
+        y=alt.Y(
+            "reference_value",
+            scale=alt.Scale(domain=(event_value_range[0], event_value_range[-1])),
+        ),
+        color=alt.ColorValue("black"),
+        tooltip=[
+            alt.Tooltip(
+                "event_start:T",
```

Comment:
You seem to repeat some tooltips (event_start, event_end) which the `base_chart` function already provides?

Reply:
I tried to add to the list of previously defined tooltips, but it seems as if we need to overwrite the whole thing.